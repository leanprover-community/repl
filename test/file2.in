{"path": "test/file2.lean", "allTactics": true}
