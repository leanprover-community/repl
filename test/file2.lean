def a : Nat := 37
def b : Nat := 37

theorem h1 : a = b := rfl

theorem h (x y z : Nat) : x + y + (z + a) = x + (z + b + y) := by
  rw [Nat.add_assoc, Nat.add_comm y, h1]
