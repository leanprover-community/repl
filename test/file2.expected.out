{"tactics":
 [{"usedConstants":
   ["Eq.mpr",
    "congrArg",
    "id",
    "a",
    "h1",
    "instHAdd",
    "b",
    "HAdd.hAdd",
    "Nat",
    "Nat.add_assoc",
    "instAddNat",
    "Eq.refl",
    "Eq",
    "Nat.add_comm"],
   "tactic": "rw [Nat.add_assoc, Nat.add_comm y, h1]",
   "proofState": 0,
   "pos": {"line": 7, "column": 2},
   "goals": "x y z : Nat\n‚ä¢ x + y + (z + a) = x + (z + b + y)",
   "endPos": {"line": 7, "column": 40}}],
 "messages":
 [{"severity": "info",
   "pos": {"line": 4, "column": 0},
   "endPos": {"line": 4, "column": 25},
   "data": "Goals accomplished!"},
  {"severity": "info",
   "pos": {"line": 6, "column": 0},
   "endPos": {"line": 7, "column": 40},
   "data": "Goals accomplished!"}],
 "env": 0}

